src/main/java/com/conveyal/analysis/controllers/OpportunityDatasetController.java

```diff
@@ -171,47 +174,28 @@ private OpportunityDatasetUploadStatus downloadLODES(Request req, Response res)
      * Given a list of new PointSets, serialize each PointSet and save it to S3, then create a metadata object about
      * that PointSet and store it in Mongo.
      */
-    private List<OpportunityDataset> createDatasetsFromPointSets(String email,
-                                                                       String accessGroup,
-                                                                       String sourceName,
-                                                                       String sourceId,
-                                                                       String regionId,
-                                                                       OpportunityDatasetUploadStatus status,
-                                                                       List<? extends PointSet> pointSets) {
+    private void updateAndStoreDatasets (SpatialResource source,
+                                         OpportunityDatasetUploadStatus status,
+                                         List<? extends PointSet> pointSets) {
         status.status = Status.UPLOADING;
         status.totalGrids = pointSets.size();
-
         // Create an OpportunityDataset holding some metadata about each PointSet (Grid or FreeForm).
         final List<OpportunityDataset> datasets = new ArrayList<>();
         for (PointSet pointSet : pointSets) {
-
-            // Make new PointSet metadata objects.
-            // Unfortunately we can't pull this step out into a method because there are so many parameters.
-            // Some of that metadata could be consolidated e.g. user email and access group.
             OpportunityDataset dataset = new OpportunityDataset();
-            dataset.sourceName = sourceName;
-            dataset.sourceId = sourceId;
+            dataset.sourceName = source.name;
+            dataset.sourceId = source._id.toString();
+            dataset.createdBy = source.createdBy;
```

Comment:
If we allow uploading data sources, then creating opportunity data sets from them later, then the createdBy may not be the same person as on the source right?